### PR TITLE
add sitecore xp 9.2, sxa 1.9.0 and jss 12.0.0

### PR DESCRIPTION
**Pull request by izharikov**

### CODE REVIEW

**Comment by outruller** [comment]
Hi, @izharikov!
Thanks for your PR!

It will take a couple of days to go thru your changes.
And I hope that you won't be too upset if we will ask a couple of questions or request some changes. :)

Thanks again for your input!

**Comment by izharikov** [comment]
@outruller, sure.

Fill free for any suggestions. I'll try to review and to resolve them.

File: src/components/sitecore/chef/cookbooks/scp_sitecore_modules/recipes/jss_1200_sc92.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Install NodeJS LTS
+scp_sitecore_modules_install_nodejs_lts 'Install NodeJS LTS' do
+end
+
+# Install Sitecore JavaScript Services
+scp_sitecore_modules_install_jss 'Install Sitecore JavaScript Services' do
+  options Chef::Mixin::DeepMerge.deep_merge(node['scp_sitecore_modules']['common_sc92'], node['scp_sitecore_modules']['jss_1200_sc92'])
```
Comment (outruller) [inline src/components/sitecore/chef/cookbooks/scp_sitecore_modules/recipes/jss_1200_sc92.rb]: `common_sc92` => `common`?

File: src/components/sitecore/chef/cookbooks/scp_sitecore_modules/recipes/sxa_190_sc92.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Install Sitecore Experience Accelerator
+scp_sitecore_modules_install_sxa 'Install Sitecore Experience Accelerator' do
+  options Chef::Mixin::DeepMerge.deep_merge(node['scp_sitecore_modules']['common_sc92'], node['scp_sitecore_modules']['sxa_190_sc92'])
```
Comment (outruller) [inline src/components/sitecore/chef/cookbooks/scp_sitecore_modules/recipes/sxa_190_sc92.rb]: `common_sc92` => `common`?

File: src/components/sitecore/chef/cookbooks/scp_sitecore_modules/recipes/sxa_190_sc92.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  secrets node['scp_sitecore_common']['secrets']
+  action :install
+end
+
+# Add SOLR cores for Sitecore Experience Accelerator
+scp_sitecore_modules_install_sxa 'Add SOLR cores for Sitecore Experience Accelerator' do
+  options Chef::Mixin::DeepMerge.deep_merge(node['scp_sitecore_modules']['common_sc92'], node['scp_sitecore_modules']['sxa_190_sc92'])
```
Comment (outruller) [inline src/components/sitecore/chef/cookbooks/scp_sitecore_modules/recipes/sxa_190_sc92.rb]: `common_sc92` => `common`?

File: src/components/sitecore/chef/cookbooks/scp_sitecore_modules/attributes/common_sc92.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+default['scp_sitecore_modules']['common_sc92'] = {
```
Comment (outruller) [inline src/components/sitecore/chef/cookbooks/scp_sitecore_modules/attributes/common_sc92.rb]: delete `common_sc92` and use `common` instead?

File: src/components/iis/chef/cookbooks/scp_iis/resources/server.rb [hunk 5]
```diff
@@ -3,6 +3,13 @@
 default_action :install
 
 action :install do
+  scp_windows_powershell_script_elevated "Installing Web-Server options" do
+    code <<-EOH
+      Install-WindowsFeature -name Web-Server -IncludeManagementTools
```
[the remark this answers is not in the document]
Comment (izharikov) [reply src/components/iis/chef/cookbooks/scp_iis/resources/server.rb]: It was done for local installation (for some unknown reasons only with this section IIS features were installed on my VM). Removed.

File: src/containers/w/packer/builders/virtualbox/template.json [hunk 6]
```diff
@@ -8,7 +8,7 @@
     "virtualbox_winrm_username": "vagrant",
     "virtualbox_winrm_password": "vagrant",
     "virtualbox_winrm_timeout": "2h",
-    "virtualbox_guest_additions_mode": "disable",
+    "virtualbox_guest_additions_mode": "upload",
```
[the remark this answers is not in the document]
Comment (izharikov) [reply src/containers/w/packer/builders/virtualbox/template.json]: It was also local modification for testing purpose, removed.

File: src/components/windows/chef/cookbooks/scp_windows/resources/updates.rb [hunk 7]
```diff
@@ -67,10 +67,10 @@
...
   scp_windows_powershell_script_elevated 'Clean up Updates' do
     code <<-EOH
-      DISM.exe /Online /Cleanup-Image /AnalyzeComponentStore
-      DISM.exe /Online /Cleanup-Image /StartComponentCleanup
-      DISM.exe /Online /Cleanup-Image /StartComponentCleanup /ResetBase
-      DISM.exe /Online /Cleanup-Image /AnalyzeComponentStore
+      # DISM.exe /Online /Cleanup-Image /AnalyzeComponentStore
+      # DISM.exe /Online /Cleanup-Image /StartComponentCleanup
+      # DISM.exe /Online /Cleanup-Image /StartComponentCleanup /ResetBase
+      # DISM.exe /Online /Cleanup-Image /AnalyzeComponentStore
```
[the remark this answers is not in the document]
Comment (izharikov) [reply src/components/windows/chef/cookbooks/scp_windows/resources/updates.rb]: Again, local modification, as in this section I got infinite load (don't know why). Un-commented back.

File: src/components/commerce/chef/cookbooks/scp_commerce/recipes/901.rb [hunk 8]
```diff
@@ -1,5 +1,5 @@
 scp_commerce_901 '' do
   options Chef::Mixin::DeepMerge.deep_merge(node['scp_commerce']['install'], node['scp_commerce']['901'])
-  secrets node['scp_commerce']['secrets']
+  secrets node['scp_sitecore_common']['secrets']
```
[the remark this answers is not in the document]
Comment (izharikov) [reply src/components/commerce/chef/cookbooks/scp_commerce/recipes/901.rb]: Added section to `README.MD` and changes to `secrets node['scp_commerce']['secrets']`.

**Comment by izharikov** [comment]
@outruller, could you please review changes again, I tried to solve all of them.

**Comment by outruller** [comment]
I did the testing and packed new boxes - everything is great except new SPE commit, as soon as it will be removed - I will complete this PR.
Thanks!

**Comment by izharikov** [comment]
@outruller , SPE commit is required, cause sitecore 9.2 requires different installation of SPE. 
On [SXA 1.9.0 downloads page](https://dev.sitecore.net/Downloads/Sitecore_Experience_Accelerator/19/Sitecore_Experience_Accelerator_190.aspx):
different version of SPE
Without this commit, installation of SXA is invalid due wrong version of SPE and it leads to missing `Create Tenant` or `Create JSS Tenant` options missing on insert options of `sitecore\content` item. So, there is no full support of SXA without this commit in generated VMs.

**Comment by izharikov** [comment]
@outruller , these packages has different size.
Difference between SPE 9.1 and 9.2
(also, Cognifide.PowerShell.dll has different size). And again, with previous SPE some SXA functionality is missing. I'm absolutely sure, that we need this commit to install valid version of SPE.

**Comment by outruller** [comment]
> @outruller , these packages has different size.
> Difference between SPE 9.1 and 9.2
> (also, Cognifide.PowerShell.dll has different size). And again, with previous SPE some SXA functionality is missing. I'm absolutely sure, that we need this commit to install valid version of SPE.

My apologies, you are absolutely correct - they are different. Thanks for noticing it!
image

**Pull request merged**